layout/layout.go

```diff
@@ -0,0 +1,740 @@
+		parts := strings.Split(envVar, "=")
+		if parts[0] == key {
+			return parts[1], nil
+		}
+	}
+	return "", nil
+}
+
+func (i *Image) WorkingDir() (string, error) {
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return "", errors.Wrapf(err, "getting config file for image at path %q", i.path)
+	}
+	if cfg == nil {
+		return "", fmt.Errorf("missing config for image at path %q", i.path)
+	}
+	return cfg.Config.WorkingDir, nil
+}
+
+func (i *Image) Entrypoint() ([]string, error) {
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting config file for image at path %q", i.path)
+	}
+	if cfg == nil {
+		return nil, fmt.Errorf("missing config for image at path %q", i.path)
+	}
+	return cfg.Config.Entrypoint, nil
+}
+
+func (i *Image) OS() (string, error) {
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return "", errors.Wrapf(err, "getting config file for image at path %q", i.path)
+	}
+	if cfg == nil {
+		return "", fmt.Errorf("missing config for image at path %q", i.path)
+	}
+	if cfg.OS == "" {
+		return "", fmt.Errorf("missing OS for image at path %q", i.path)
+	}
+	return cfg.OS, nil
+}
+
+func (i *Image) OSVersion() (string, error) {
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return "", errors.Wrapf(err, "getting config file for image at path %q", i.path)
+	}
+	if cfg == nil {
+		return "", fmt.Errorf("missing config for image at path %q", i.path)
+	}
+	return cfg.OSVersion, nil
+}
+
+func (i *Image) Architecture() (string, error) {
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return "", errors.Wrapf(err, "getting config file for image at path %q", i.path)
+	}
+	if cfg == nil {
+		return "", fmt.Errorf("missing config for image at path %q", i.path)
+	}
+	if cfg.Architecture == "" {
+		return "", fmt.Errorf("missing Architecture for image at path %q", i.path)
+	}
+	return cfg.Architecture, nil
+}
+
+func (i *Image) Name() string {
+	return i.path
+}
+
+func (i *Image) Rename(name string) {
+	i.path = name
+}
+
+// Found tells whether the image exists in the repository by `Name()`.
+func (i *Image) Found() bool {
+	return imageExists(i.path)
+}
+
+// Identifier
+// Each image's ID is given by the SHA256 hash of its configuration JSON. It is represented as a hexadecimal encoding of 256 bits,
+// e.g., sha256:a9561eb1b190625c9adb5a9513e72c4dedafc1cb2d4c5236c9a6957ec7dfd5a9.
+func (i *Image) Identifier() (imgutil.Identifier, error) {
+	configFile, err := i.Image.ConfigFile()
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting identifier for image at path %q", i.path)
+	}
+	return newIDIdentifier(configFile)
+}
+
+func (i *Image) CreatedAt() (time.Time, error) {
+	configFile, err := i.Image.ConfigFile()
+	if err != nil {
+		return time.Time{}, errors.Wrapf(err, "getting createdAt time for image at path %q", i.path)
+	}
+	return configFile.Created.UTC(), nil
+}
+
+func (i *Image) Rebase(s string, image imgutil.Image) error {
+	return nil
```

Comment:
Let's return a not-implemented error instead, that way callers can be alerted

Reply:
Done